### ISSUE TITLE
What's next

### ISSUE THREAD

**Issue by Swiip** [FountainJS/generator-fountain-webapp#48]
Some people asked us about the next focuses and goals of the project. Here ours list as it is. We don't really order things and pick next subjects part because of the trends, part because of our will :)

Of course, any contribution on one of these subjects will be more than welcome.

To finish
- [x] Codelab http://yeoman.io/codelab 
- [x] TodoMVC
- [ ] Isomorphic pre-rendering post

Missing
- [x] Routing
- [ ] UI frameworks?
- [ ] More blog posts

New generators
- Frameworks
  - [ ] Polymer
  - [ ] Ember 2
  - [x] VueJS 2
  - [ ] Backbone
- Module system
  - [ ] Browserify
- CSS
  - [ ] PostCSS / CSSNext
  - [x] Stylus

Web site
- [ ] Home per framework
- [ ] Pre - generated archives

Refactoring
- [ ] Renaming generators eslint and tslint
- [ ] Merging all Techs css

Improving
- [ ] Ordering for json2js
- [ ] Comment for json2js
- [x] Improve Unit Test coverage
- [x] Move integration tests on TodoMVC
- [x] Angular 2 external template
- [x] Using yo-rc instead of this.options

Proof of concetps
- [ ] Isomorphism
- [ ] Backend
- [ ] Mobile

**Comment by qlonik** [comment]
Hello,

Are there any plans to bring protractor tests back into at least angular 1 from gulp-angular? I'm not sure if they even work in angular 2, but it would be nice to have them in angular 1.

On the side note, could you please share outline or couple ideas that you have about routing? Maybe some people who really need that could implement and send a pull request?

Thanks!

**Comment by Delapouite** [comment]
Stylus support would be really nice! I'll try to make a PR, but first I'll try to understand the full architecture of the nested projects.

The first step is to work on https://github.com/FountainJS/generator-fountain-gulp right?

**Comment by micaelmbagira** [comment]
@Delapouite Thanks for contributing, yes it would be nice to have stylus support!
- So, right, the styles compilation is there https://github.com/FountainJS/generator-fountain-gulp/blob/master/generators/app/templates/gulp_tasks/styles.js
- For Webpack, it's slightly different, we use css/sass/less loaders https://github.com/FountainJS/generator-fountain-webpack/blob/master/generators/app/conf.js#L109-L116 and https://github.com/FountainJS/generator-fountain-webpack/blob/master/generators/app/index.js#L39-L51
- Then you can add `.styl` files in templates folders such as https://github.com/FountainJS/generator-fountain-angular1/tree/master/generators/techs/modules/templates/src, there is one for each sample app we generate so `techs`, `hello` and `todoMVC` and for each framework ;)

**Comment by Delapouite** [comment]
Thanks for the pointers micael. First PRs incoming : 
- gulp https://github.com/FountainJS/generator-fountain-gulp/pull/37
- webpack https://github.com/FountainJS/generator-fountain-webpack/pull/32
- react https://github.com/FountainJS/generator-fountain-react/pull/44
- angular https://github.com/FountainJS/generator-fountain-angular1/pull/42
- angular2 https://github.com/FountainJS/generator-fountain-angular2/pull/56
- fountain https://github.com/FountainJS/fountain-generator/pull/30
- website https://github.com/FountainJS/fountainjs.github.io/pull/12

Maybe we should open a new issue dedicated to stylus to avoid spamming this one

**Comment by ghost** [comment]
Is there any plans to support MeteorJS?

**Comment by JimLynchCodes** [comment]
How about Redux? That's becoming the de facto way to manage state for Angular 1, Angular 2, and React. It would awesome to have one of the quesions be, "Do you want to have Redux already in your project and set up for you?". :)

**Comment by micaelmbagira** [comment]
Hi @JimTheMan, if you select the TodoMVC option you will have Redux set up with React and ngrx/store (one implementation of Flux pattern) set up with Angular 2

**Comment by zoamel** [comment]
Yeah but having ngRedux in Angular 1.x would also be beneficial, cause there are not many tutorials about how to integrate this approach with first angular.

**Comment by lhas** [comment]
Great job here, folks! Thanks for this generator. Pretty useful! :+1:

**Comment by JimLynchCodes** [comment]
Welcome! Now that you've joined the club will you ever use another generator? ;)

**Comment by vinicius73** [comment]
Looking forward to version vue js :metal: !

**Comment by fuchao2012** [comment]
waiting for GMD

**Comment by chadfurman** [comment]
Mithril: A very cool, lightweight JS framework.   http://mithril.js.org/
Polymer: Setting up a process / folder structure to make building with Polymer easy will be appreciated

Maybe a poll to determine the highest priority for your framework integrations?

**Comment by mhadaily** [comment]
Hi, How may I help with Ember 2? I am ready to contribute if anybody is not working on that.
